torch/testing/_internal/common_methods_invocations.py

```diff
@@ -422,6 +422,64 @@ def sample_inputs_linalg_norm(op_info, device, dtype, requires_grad):
                             dim=(0, 1))))
         return inputs
 
+def sample_inputs_linalg_vector_norm(op_info, device, dtype, requires_grad):
+    test_sizes = [
+        (S,),
+        (S, S),
+        (S, S, S),
+        (S, S, S, S),
+        (0,),
+        (0, 0),
+        (S, 0),
+        (0, S),
+        (0, S, S),
+        (S, 0, S),
+        (0, 0, 0),
+    ]
+    ords = (None, 0, 0.5, 1, 2, 3.5, inf, -0.5, -1, -2, -3.5, -inf)
+    inputs = []
+    is_dtype_half = dtype in [torch.float16, torch.bfloat16]
+
+    def get_all_unique_dim_args(ndim):
+        dim_args = [None]
+        all_dims = list(range(ndim))
+        for num_reduction_dims in range(1, len(test_size) + 1):
+            for dim in combinations(all_dims, num_reduction_dims):
+                dim_args.append(dim)
+        return dim_args
+
+    for test_size in test_sizes:
+        dim_args = get_all_unique_dim_args(len(test_size))
+        for keepdim in [False, True]:
+            for dim in dim_args:
+                inputs.append(SampleInput(
+                    make_tensor(
+                        test_size, device, dtype, low=None, high=None,
+                        requires_grad=requires_grad),
+                    kwargs=dict(
+                        keepdim=keepdim,
+                        dim=dim)))
+
+            for ord in ords:
+                # TODO: remove this check when `max` is implemented for
+                #       float16 and bfloat16. Issue:
+                #       https://github.com/pytorch/pytorch/issues/50790
+                if is_dtype_half and ord in [inf, -inf]:
+                    continue
+
+                for dim in dim_args:
```

Comment:
I like how thorough this test is and appreciate its structure, but I think this only needs to append one set of these SampleInputs? I also think this can enumerate the product of keepdim, dim_args, and ords to avoid nesting.